.github/workflows/build-and-test.yml

```diff
@@ -90,20 +101,24 @@ jobs:
 # Build tag push the image after a merge to develop
       -
         name: Build, Tag, Push
-        uses: docker/build-push-action@v3
+        uses: docker/build-push-action@v4
         if: github.ref == 'refs/heads/develop'
         with:
           push: true
           tags: ${{ env.IMAGE_REPOSITORY }}:develop-latest
           build-args: |
               REVISION_HASH=${GITHUB_SHA}
+          cache-from: type=registry,ref=${{ env.IMAGE_CACHE_REPOSITORY }}:cache
+          cache-to: type=registry,ref=${{ env.IMAGE_CACHE_REPOSITORY }}:cache
```

Comment:
I'm confused as to why these are cached? A build, tag, and push is frequently going to be a unique output. What is the purpose of caching this step?

When I am triggering an image build, I almost always want the build to be fresh and unique so as to ensure that its dependencies are installed for any package.json changes, the build is fresh, and the release is tagged appropriately.

Is this to re-use the build that was generated during the testing/continuous-integration phases?

Reply:
We currently don't have a docker image build in the CI workflow, but i could add one strictly for caching purposes. The idea with the image cache in the build, tag, push step is that it uses docker image layers that remain unchanged instead of building the entire image every time.

Reply:
You bring up a good point, though. The image builds are only happening on `develop` and release tags, where caching is less important due to the fact that developers aren't waiting on those builds like they would be during active development while running the CI/testing workflow, and where waiting a little longer for a fresh build might actually be preferable. 

I suppose a better solution here would be to containerize the CI workflow and cache those builds to help speed up the development process.

Reply:
I like your ideas around containerizing and caching the CI/testing workflow builds. I would be in favour of this as long as we could guarantee the dependencies (`npm i`) and resultant build would match the expected output for a develop/release branch.

As an alternative, it might be worth categorizing the cache based on branch (i.e. `develop`, `release-x.y.z`, etc) so that there aren't a chance for something to draw from a cached object other than its past selves? This still has the same issues of needing to ensure dependencies and build are correct every time, but would at least avoid having to containerize the whole process right now.